bytecomp/translcore.ml

```diff
@@ -1218,16 +1218,16 @@ and transl_apply ?(should_be_tailcall=false) ?(inlined = Default_inline)
      : Lambda.lambda)
 
 and transl_function loc untuplify_fn repr partial param cases =
-  match cases with
+  match cases, partial with
     [{c_lhs=pat; c_guard=None;
       c_rhs={exp_desc = Texp_function { arg_label = _; param = param'; cases;
-        partial = partial'; }} as exp}]
-    when Parmatch.fluid pat ->
+        partial = partial'; }} as exp}], Total
+    when Parmatch.inactive pat ->
```

Comment:
Why did you decide to change the way (ir)refutability/partiality is checked in this test? (I'm fine with moving away with the fairly unclear name "fluid".) Is there a difference in meaning between `Parmatch.irrefutable` and this new test?

Reply:
Cf Leo's note in the PR description:

> To try and avoid conflicts I've made it on top of #1195, so when reviewing this change only the second commit is relevant.